science/openmpi/Portfile

```diff
@@ -185,9 +185,11 @@ if {${cname} in ${clist_unsupported}} {
         set cname                   mp
     }
 
-    # Force buildbot to skip this since openmpi depends on hwloc and that would
-    # be different for a user's machine
-    archive_sites
+    if {[lsearch -exact {mp llvm clang} ${cname}] != -1} {
```

Comment:
Reviewers: Note that we don't presently have an openmpi subport named `openmpi-llvm`. However, to be consistent with mpich, I'm using the same logic when determining which buildbot binaries are not to be used.

Reply:
If this line need to be kept in sync with that in another port please add a clear comment here *and* in that other port detailing this.

Reply:
Chris, technically the line doesn't have to be kept in-sync with mpich. I'm simply using the same logic for consistency.

I'm willing to add a comment though, as part of a separate commit. But I'd prefer to wait until I'm granted commit access.

Can we merge this as-is for now?

Reply:
Chris, I didn't do a very good job of explaining this before, so let me elaborate further.

In short, there is a _significant_ amount of logic that is duplicated between ports `openmpi` and `mpich`. Some of it was refactored to the `mpi` portgroup long before my arrival, but there's still plenty of duplication between the two. And there are just enough differences - such as the target compilers supported - making logic sharing more difficult.

Nonetheless, there is a long-term opportunity to share more logic, and simplify both ports, via the `mpi` portgroup. I'll likely take a look at some point during the coming months, unless someone beats me to it. And I'd certainly welcome anyone who wishes to collaborate on it.

Regardless, here's the point: If we start adding comments to logic duplicated between the two, that would relate to far more than just this block. And we'd be littering the two ports with comments, which simply isn't practical.

Instead, it makes more sense to point out the fact that the two share quite a bit of logic, and emphasize the need to keep both in-sync where it makes sense. And indeed, my latest commit includes just such a comment block, on line 10 above.

I'll be adding an identical comment block to the top of 'mpich' as part of a separate PR, as some of fixes here now need to be applied there too. But given that the CI jobs have a great chance of timeout for one of these ports alone, including both in one PR is pointless.

Does all of that make sense?

Reply:
PR for `mpich`:

[10879 - mpich: ensure target compilers are added to depends_lib](https://github.com/macports/macports-ports/pull/10879)

Reply:
Proposed new portgroup, allowing us to share logic between `openmpi` and `mpich`, tracked by ticket:

[62807 - openmpi/mpich: refactor duplicated logic, and simplify ports, via new portgroup](https://trac.macports.org/ticket/62807)